src/codegen/compile.ml

```diff
@@ -1030,12 +1057,42 @@ module Stack = struct
     G.i (Binary (Wasm.Values.I32 I32Op.Add)) ^^
     set_stack_ptr env
 
+  (* TODO: why not just remember and reset the stack pointer, instead of calling free_words? Also below *)
```

Comment:
Don’t remember, maybe no reason at all.

Reply:
If we do this, worth checking that the stackpointer is as expected afterwards? At least in debug mode?